python-checks/src/main/java/org/sonar/python/checks/UnusedLocalVariableCheck.java

```diff
@@ -138,10 +141,30 @@ public PreciseIssue createIssue(SubscriptionContext ctx, Symbol symbol, Usage us
     } else {
       var issue = ctx.addIssue(usage.tree(), String.format(MESSAGE, symbol.name()));
       createExceptClauseQuickFix(usage, issue);
+      createAssignmentQuickFix(usage, issue);
       return issue;
     }
   }
 
+  private static void createAssignmentQuickFix(Usage usage, PreciseIssue issue) {
+    if (usage.kind().equals(Usage.Kind.ASSIGNMENT_LHS)) {
+      Statement assignmentStatement = ((Statement) TreeUtils.firstAncestorOfKind(usage.tree(), Kind.ASSIGNMENT_STMT, Kind.ANNOTATED_ASSIGNMENT));
+
+      Optional.ofNullable(assignmentStatement).filter(stmt -> stmt.is(Kind.ASSIGNMENT_STMT)).map(AssignmentStatement.class::cast).ifPresent(stmt -> {
+        PythonQuickFix quickFix = PythonQuickFix.newQuickFix(ASSIGNMENT_QUICK_FIX_MESSAGE,
+          TextEditUtils.removeUntil(usage.tree(), stmt.assignedValue().firstToken()));
+        issue.addQuickFix(quickFix);
+      });
+
+      Optional.ofNullable(assignmentStatement).filter(stmt -> stmt.is(Kind.ANNOTATED_ASSIGNMENT)).map(AnnotatedAssignment.class::cast)
+        .map(AnnotatedAssignment::assignedValue).ifPresent(stmt -> {
```

Comment:
As mentioned in the previous review, I think that having a test case similar to:

```
def annotated_assignment_null_value():
    value: str
    value = "hello"
```

is important.

The lack of this test case led you to assume `AnnotatedAssignment::assignedValue` could never return `null` when reaching this point (although it can). Here, the `Optional` chain is hiding the fact we are missing some attainable coverage (I'm not saying we shouldn't use an `Optional` here, just that we shouldn't use it for the wrong reasons).

Namely, in this case, an issue would be raised but no quick fix will be offered (we can have a test in the regular check test as well as one in the quick fix test to show this).